pkg/scheduler/framework/plugins/podtopologyspread/filtering_test.go

```diff
@@ -1220,13 +1221,14 @@ func TestPreFilterStateAddPod(t *testing.T) {
 	zoneConstraint := nodeConstraint
 	zoneConstraint.TopologyKey = "zone"
 	tests := []struct {
-		name         string
-		preemptor    *v1.Pod
-		addedPod     *v1.Pod
-		existingPods []*v1.Pod
-		nodeIdx      int // denotes which node 'addedPod' belongs to
-		nodes        []*v1.Node
-		want         *preFilterState
+		name                       string
+		preemptor                  *v1.Pod
+		addedPod                   *v1.Pod
+		existingPods               []*v1.Pod
+		nodeIdx                    int // denotes which node 'addedPod' belongs to
+		nodes                      []*v1.Node
+		want                       *preFilterState
+		enableNodeInclustionPolicy bool
```

Comment:
this makes me think that we need similar unit tests for TestPreFilterRemovePod (which could be done in a follow up).

Unless, we have a similar bug in RemovePod, then we should fix it in this PR.

/lgtm
/approve
/hold
to verify if RemovePod needs to be fixed.

Reply:
Both `AddPod` and `RemovePod` will call `updateWithPod`, so it's fixed at the same time. I'll follow a new PR to cover the remove testcases.

Reply:
Thanks!

/hold cancel